nixos/modules/services/web-apps/mealie.nix

```diff
@@ -0,0 +1,108 @@
+{ config, lib, pkgs, ...}:
+let
+  cfg = config.services.mealie;
+  pkg = pkgs.mealie;
+in
+{
+  options.services.mealie = {
+    enable = lib.mkEnableOption "Mealie, a recipe manager and meal planner";
+
+    # TODO allow_signup
+
+    host = lib.mkOption {
+      type = lib.types.str;
+      default = "localhost";
+      description = "Host on which creating the URL on the service";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9000;
+      description = "Port on which to serve the Mealie service";
+    };
+
+    protocol = lib.mkOption {
+      type = lib.types.str;
+      default = "http";
+      description = "Protocol to use to serve the service";
+    };
+
+    logLevel = lib.mkOption {
+      type = lib.types.str;
+      default = "INFO";
+      description = "Log level to use for this service";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.mealie = {
+      description = "Mealie, a self hosted recipe manager and meal planner";
+      after = [
+        "network.target"
+        "network-online.target"
+      ];
+      wantedBy = [
+        "multi-user.target"
+      ];
+
+      environment = {
+        PYTHONPATH = "${pkg.pythonPath}:${pkg}/lib/${pkg.python.libPrefix}/site-packages";
+        STATIC_FILES = "${pkg.frontend}";
+
+        # Data directory defined by systemd service
+        MEALIE_LOG_FILE = "/var/log/mealie/mealie.log";
+        ALEMBIC_CONFIG_FPATH="/var/lib/mealie/alembic.ini";
+
+        # Default configs
+        DEFAULT_GROUP="Home";
+        DEFAULT_EMAIL="changeme@example.com";
+        DEFAULT_PASSWORD="MyPassword";
+        PRODUCTION = "true";
+        API_DOCS = "False";
+        DB_ENGINE = "sqlite";
+        TOKEN_TIME="24";
+
+        # Configuration from module
+        LOG_LEVEL = cfg.logLevel;
+        API_PORT = builtins.toString cfg.port;
+        BASE_URL = "${cfg.protocol}://${cfg.host}:${builtins.toString cfg.port}";
+
+        # TODO  Additionnal config
+        # ALLOW_SIGNUP = cfg.allow_signup;
+        # POSTGRES_USER=mealie
+        # POSTGRES_PASSWORD=mealie
+        # POSTGRES_SERVER=postgres
+        # POSTGRES_PORT=5432
+        # POSTGRES_DB=mealie
+        # LDAP_AUTH_ENABLED=False
+        # LDAP_SERVER_URL=""
+        # LDAP_TLS_INSECURE=False
+        # LDAP_TLS_CACERTFILE=
+        # LDAP_ENABLE_STARTTLS=False
+        # LDAP_BASE_DN=""
+        # LDAP_QUERY_BIND=""
+        # LDAP_QUERY_PASSWORD=""
+        # LDAP_USER_FILTER="(&(|({id_attribute}={input})({mail_attribute}={input}))(objectClass=person))"
+
+        # LDAP_ADMIN_FILTER=""
+        # LDAP_ID_ATTRIBUTE=uid
+        # LDAP_NAME_ATTRIBUTE=name
+        # LDAP_MAIL_ATTRIBUTE=mail
```

Comment:
Rather than try to introduce 1001 options, I would add an `extraConfig` option similar to `paperless`.

You can them do something like:

```
services.mealie.extraConfig = {
    DB_ENGINE = lib.mkDefault "sqlite";
    API_DOCS = lib.mkDefault "False";
    TOKEN_TIME = lib.mkDefault "24";
    PRODUCTION = lib.mkDefault "true";
    # Arguably, `logLevel` isn't very useful as an option, it could just 
    LOG_LEVEL = cfg.logLevel;
};
```

BTW: different casing between `False` and `true` looks weird.

Reply:
Ok I tried to do it like this, it should be much better now !
Also, I noticed the case difference between booleans, but frankly I'm too scared to attempt to put everything in a case, as it may screw up env var "parsing" inside the Python code